### ISSUE TITLE
[Relay Mining] Maintain per service difficulty on the relayer

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/poktroll#705]
## Objective

Solve an overlooked issue whereby the relay miner uses the global parameter

## Origin Document

A specific issue was identified [here](https://github.com/pokt-network/poktroll/blob/0049068b809dca0ae668be09ee18fa97e3e73b0c/pkg/relayer/miner/miner.go#L32):

![Screenshot 2024-07-25 at 12 01 59 PM](https://github.com/user-attachments/assets/fd231fc9-52ef-4871-a22d-36de5b378d9d)

This is part of a series of other TODOs that can be found [here](https://github.com/search?q=repo%3Apokt-network%2Fpoktroll%20TODO_FOLLOWUP(%40olshansk&type=code) from `TODO_FOLLOWUP(@olshansk, #690)`

## Deliverables

- [ ] Ensure the relay miner is using the difficulty per service
- [ ] Remove the `relay_difficulty_target_hash` parameter
- [ ] Search for all instances of `#705` in the codebase and tend to them
- [ ] Search for all instances of `TODO_FOLLOWUP(@olshansk, #690)` and tend to them if applicable

## Non-goals / Non-deliverables

- Tending to all of the TODO_FOLLOWUPS

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @Olshansk 
**Co-Owners**: @adshmh @red-0ne

**Comment by Olshansk** [comment]
@adshmh Just wanted to call out that there will be a few references to the the ticket #705 in the repo after #708 is merged in. Please review the updated deliverables here before jumping in.

**Comment by red-0ne** [comment]
@adshmh I think this is solved by #745, could you confirm it before we close this issue?

**Comment by red-0ne** [comment]
We actually need to use per-service difficulty on-chain too, I would keep this opened.

**Comment by Olshansk** [comment]
I thought I saw us doing this (or added a TODO). @adshmh could you confirm?

**Comment by red-0ne** [comment]
This is implemented on chain too.
